src/inference/src/dev/make_tensor.cpp

```diff
@@ -205,6 +205,10 @@ class AllocatedTensor : public ViewTensor {
                          OPENVINO_ASSERT(allocator, "Allocator was not initialized");
                          const auto byte_size = element::get_memory_size(element_type, shape_size(shape));
                          auto data = const_cast<Allocator&>(allocator).allocate(byte_size);
+                         if (byte_size != 0 && data == nullptr) {
+                             OPENVINO_THROW("Failed to allocate memory");
+                         }
```

Comment:
please, use OPENVINO_ASSERT where you can put condition inside

Reply:
Afaik, many projects follow a convention similar to this: assertion only for errors that cannot happen, usually for catching internal errors related to the project's impelentation, and throws for rutime errors, e.g. for OOM cases. That's what I've noted in [this ticket](https://jira.devtools.intel.com/browse/EISW-121701) and you may find such recommendations all over the internet (e.g. [here](https://stackoverflow.com/a/409981)).

I don't mind changing to an assert, but just for my knowledge: does OV not follow such convention?

Reply:
In OpenVINO assert is the same as exception, but it's just written using a single line w/o extra `if` block

Reply:
Just expand error message [here](https://github.com/openvinotoolkit/openvino/pull/25711/files#diff-71abf35fb7cfe65cced27683882b85e9835bf13aaab1f2d26cb00d687c35739bR56), where origin assert message occurs.

There is no point to check data pointer it will throw an error in `allocate`  at line above.